### ISSUE TITLE
DNS-Challenge Exception: "Error in Go routine: runtime error: invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by bastianwegge** [containous/traefik#3989]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

- Setup a cloud instance with your cloud provider (in my case it's cloud.hetzner.de)
- Register a Domain with DNSimple
- Configure the DNS-Records to point to your server (*.mydomain.com / mydomain.com)
- Initialize it with Docker-Machine
```bash
docker-machine create --driver generic \
                      --generic-ip-address 1.2.3.4 \
                      --generic-ssh-user root \
                      --generic-ssh-key ~/.ssh/id_rsa <machine_name>
```
- switch your local docker-client to the remote `eval $(docker-machine env <machine_name>)`
- create a `traefik.toml` (please take this out of the toml section below)
- for this to work we need a minimal html web-server, create a `Dockerfile` and an `index.html`
```
# Dockerfile for ~6 MB Web-Server 🌈
FROM pierrezemb/gostatic
COPY index.html /srv/http/index.html

# index.html
<html><body><h1>hello world</h1></body></html>
```

- create a `docker-compose.yml`
```yml
version: '2'

services:
  traefik:
    image: traefik:1.7.2
    command: --docker --api
    restart: always
    ports:
      - 80:80
      - 443:443
    networks:
      - web
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/traefik/traefik.toml:/traefik.toml
      - /opt/traefik/acme.json:/acme.json
    container_name: traefik
    environment:
      DNSIMPLE_OAUTH_TOKEN: <YOUR_DNSIMPLE_AUTH_TOKEN>
      DNSIMPLE_BASE_URL: https://api.dnsimple.com
    labels:
      traefik.backend: "traefik"
      traefik.frontend.rule: "Host:traefik.mydomain.com"
      traefik.frontend.priority: "2"
      traefik.port: "8080"
  landingpage:
    build: .
    networks:
      - web
    labels:
      traefik.mangeno.frontend.rule: HostRegexp:{subdomain:[a-zA-Z0-9]+}.mydomain.com,www.mydomain.com,mydomain.com
      traefik.mangeno.frontend.priority: 1
      traefik.mangeno.port: 8043
      traefik.docker.network: web

networks:
  web:

```
- execute the following to copy the necessary files to the remote machine and start the service
```bash
docker-machine scp -r ./traefik.toml <machine_name>:/opt/traefik/traefik.toml
docker-machine ssh <machine_name> touch /opt/traefik/acme.json && chmod 600 /opt/traefik/acme.json
docker-compose up -d
```

### What did you expect to see?

Upon starting the services, Traefik should use `Lego` to obtain a wildcard-certificate using the `dns-challenge` as described here https://docs.traefik.io/configuration/acme/#wildcard-domains .

### What did you see instead?

```
traefik    | time="2018-10-08T15:49:22Z" level=info msg="Using TOML configuration file /traefik.toml"
traefik    | time="2018-10-08T15:49:22Z" level=warning msg="Unable to use DNS challenge and HTTP challenge at the same time. Fallback to DNS challenge."
traefik    | time="2018-10-08T15:49:22Z" level=info msg="Traefik version v1.7.2 built on 2018-10-04_01:44:36PM"
traefik    | time="2018-10-08T15:49:22Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.trae
fik.io/basics/#collected-data\n"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:0xc00061b000 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Com
press:false ProxyProtocol:<nil> ForwardedHeaders:0xc000301c20} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Preparing server https &{Address::443 TLS:0xc000215200 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> C
ompress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000301c40} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting server on :80"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Preparing server traefik &{Address::8080 TLS:<nil> Redirect:<nil> Auth:0xc0005b0d50 WhitelistSourceRange:[] WhiteList:<nil
> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000301c80} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting server on :443"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting server on :8080"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting provider *docker.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersio
n\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"\",\"TLS\":null,\"ExposedByDefault\":true,\"UseBindPortIP\":false,\"SwarmMode\"
:false,\"Network\":\"\"}"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Starting provider *acme.Provider {\"Email\":\"admin@ourdomain.com\",\"ACMELogging\":false,\"CAServer\":\"https://acme-v02.ap
i.letsencrypt.org/directory\",\"Storage\":\"acme.json\",\"EntryPoint\":\"https\",\"KeyType\":\"\",\"OnHostRule\":false,\"OnDemand\":false,\"DNSChallenge\":{\"Provider\":\"dnsimple
\",\"DelayBeforeCheck\":0},\"HTTPChallenge\":null,\"TLSChallenge\":null,\"Domains\":[{\"Main\":\"*.mydomain\",\"SANs\":[\"mydomain,traefik.mydomain\"]}],\"Store\":{}}"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Testing certificate renew..."
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Server configuration reloaded on :443"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Server configuration reloaded on :8080"
traefik    | time="2018-10-08T15:49:23Z" level=info msg="Server configuration reloaded on :80"
traefik    | time="2018-10-08T15:49:23Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
traefik    | goroutine 35 [running]:
traefik    | runtime/debug.Stack(0x76b3c7, 0xc00009c280, 0x25bda3d)
traefik    |    /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
traefik    | runtime/debug.PrintStack()
traefik    |    /usr/local/go/src/runtime/debug/stack.go:16 +0x22
traefik    | github.com/containous/traefik/safe.defaultRecoverGoroutine(0x215a8c0, 0x4435a20)
traefik    |    /go/src/github.com/containous/traefik/safe/routine.go:148 +0x85
traefik    | github.com/containous/traefik/safe.GoWithRecover.func1.1(0x266a948)
traefik    |    /go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
traefik    | panic(0x215a8c0, 0x4435a20)
traefik    |    /usr/local/go/src/runtime/panic.go:513 +0x1b9
traefik    | github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/dnsimple.(*DNSProvider).Timeout(0xc0008beb30, 0x289b5a0, 0xc0008beb30)
traefik    |    /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/dnsimple/dnsimple.go:135 +0x8
traefik    | github.com/containous/traefik/provider/acme.(*Provider).getClient(0xc000214cf0, 0x0, 0x0, 0x0)
traefik    |    /go/src/github.com/containous/traefik/provider/acme/provider.go:280 +0x86e
traefik    | github.com/containous/traefik/provider/acme.(*Provider).resolveCertificate(0xc000214cf0, 0xc000886f40, 0xd, 0xc0004f3230, 0x1, 0x1, 0xf43a01, 0x0, 0x0, 0x0)
traefik    |    /go/src/github.com/containous/traefik/provider/acme/provider.go:398 +0x259
traefik    | github.com/containous/traefik/provider/acme.(*Provider).Provide.func1()
traefik    |    /go/src/github.com/containous/traefik/provider/acme/provider.go:184 +0x72
traefik    | github.com/containous/traefik/safe.GoWithRecover.func1(0x266a948, 0xc000300f80)
traefik    |    /go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
traefik    | created by github.com/containous/traefik/safe.GoWithRecover
traefik    |    /go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

Traefik Versions where this error occurs:
- `1.7.2`
- `1.7.1`

Traefik Versions where this error doesn't occur:
- `1.7.0-rc4`

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug = true

logLevel = "INFO"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
  [entryPoints.traefik]
  address = ":8080"

[api]
entryPoint = "traefik"

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "mydomain.com"
watch = true
exposedbydefault = false

[acme]
email = "admin@crossbow.de"
storage = "acme.json"
# caServer = "https://acme-staging-v02.api.letsencrypt.org/directory" # for testing purposes
entryPoint = "https"
[acme.httpChallenge]
entryPoint = "http"

[acme.dnsChallenge]
  provider = "dnsimple"

[[acme.domains]]
  main = "*.mydomain.com"
  sans = ["mydomain.com,traefik.mydomain.com"]

```

**Comment by ldez** [comment]
I made a fix for dnsimple, could you try this Docker image [ldez/traefik:dnsimple](https://hub.docker.com/r/ldez/traefik/tags/).

**Comment by bastianwegge** [comment]
@ldez it does indeed fix the Bug! Is there a chance that this might get into the next release? ❤️

**Comment by traefiker** [comment]
Closed by #3998.
